development/frameworks/ReactAppStructure.md

```diff
@@ -50,19 +49,48 @@ src
 ├── store
 ├── styles
 └── utils
-tests
-├── components
-├── containers
-├── modules
-├── pages
-├── store
-└── utils
 ```
 
-## Components
-Should store all reusable components (even if they're currently only used once!). These are ideally stateless, state should be handled in the component's container.
-
-### Naming convention
+## Components and Containers
+
+### Overview
+
+<table>
+    <thead>
+        <tr>
+            <th></th>
+            <th scope="col" style="text-align:left">Presentational Components</th>
+            <th scope="col" style="text-align:left">Container Components</th>
+        </tr>
+    </thead>
+    <tbody>
+        <tr>
+          <th scope="row" style="text-align:right">Purpose</th>
+          <td>How things look (markup, styles)</td>
+          <td>How things work (data fetching, state updates)</td>
+        </tr>
+        <tr>
+          <th scope="row" style="text-align:right">Aware of Redux</th>
+          <td>No</th>
+          <td>Yes</th>
+        </tr>
+        <tr>
+          <th scope="row" style="text-align:right">To read data</th>
+          <td>Read data from props</td>
+          <td>Subscribe to Redux state</td>
+        </tr>
+        <tr>
+          <th scope="row" style="text-align:right">To change data</th>
+          <td>Invoke callbacks from props</td>
+          <td>Dispatch Redux actions</td>
+        </tr>
+    </tbody>
+</table>
+
+### Components
+Should store all reusable components (even if they're currently only used once). These are ideally stateless, state should be handled in the component's container.
```

Comment:
Might be misunderstanding but internal or as it is sometimes called UI state should be handled in the component rather than the container.

Reply:
Yeah, it's a fair point. State should be handled in the container as much as possible so I propose changing the text to read

> These are ideally stateless, state should be handled in the component's container _as much as possible_.

Something like that? We wouldn't want a component to determine something like `isComplete` or `hasError`, but it would be acceptable for, say, a date input to store day, month, and year.